services/cms/src/utils/Gutenberg/withMentimeterInspector.tsx

```diff
@@ -0,0 +1,84 @@
+import { css } from "@emotion/css"
+import { InspectorControls } from "@wordpress/block-editor"
+import { PanelBody, TextControl } from "@wordpress/components"
+import { createHigherOrderComponent } from "@wordpress/compose"
+import { Fragment } from "@wordpress/element"
+import { useTranslation } from "react-i18next"
+
+import { baseTheme } from "../../shared-module/styles"
+
+// https://developer.wordpress.org/block-editor/reference-guides/filters/block-filters/#editor-blockedit
+const withMentimeterInspector = createHigherOrderComponent((BlockEdit) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const MentiMeterEmbed = (props: any) => {
+    const { t } = useTranslation()
+    if (props.attributes.providerNameSlug !== "mentimeter") {
+      return <BlockEdit {...props} />
+    }
+
+    function updateQueryStringParameter(url: string | undefined, key: string, value: string) {
+      if (!url) {
+        return
+      }
+      // eslint-disable-next-line i18next/no-literal-string
```

Comment:
I usually don't recommend parsing URLs by hand. This could be made more clear with something like this:

```js
const parsedUrl = new URL(url)
if (parsedUrl.searchParams.has(key) {
  parsedUrl.searchParams.delete(key)
}
 parsedUrl.searchParams.append(key, value)
return parsedUrl.toString()
```

Of course, in this case we'd also need to handle the case where the `new URL` throws an exception for an invalid URL. One workaround for that could be to disable the text field when the URL is not valid.

Reply:
As long as the frontend doesn't crash